Rubberduck.Core/UI/Inspections/InspectionResultsViewModel.cs

```diff
@@ -202,136 +209,151 @@ public IEnumerable<DisplayQuickFix> QuickFixes
+            { InspectionResultGrouping.Type, new PropertyGroupDescription("Inspection", new InspectionTypeConverter()) },
+            { InspectionResultGrouping.Name, new PropertyGroupDescription("Inspection.Name") },
+            { InspectionResultGrouping.Location, new PropertyGroupDescription("QualifiedSelection.QualifiedName") },
+            { InspectionResultGrouping.Severity, new PropertyGroupDescription("Inspection.Severity") }
+        };
 
-                if (value)
-                {
-                    Results = new ObservableCollection<IInspectionResult>(
-                            Results.OrderBy(o => o.Inspection.InspectionType)
-                                .ThenBy(t => t.Inspection.Name)
-                                .ThenBy(t => t.QualifiedSelection.QualifiedName.Name)
-                                .ThenBy(t => t.QualifiedSelection.Selection.StartLine)
-                                .ThenBy(t => t.QualifiedSelection.Selection.StartColumn)
-                                .ToList());
-
-                }
+        private void SetGrouping(InspectionResultGrouping grouping)
+        {
+            Results.GroupDescriptions.Clear();
+            Results.GroupDescriptions.Add(GroupDescriptions[grouping]);
+            Results.Refresh();
 
-                _groupByInspectionType = value;
-                OnPropertyChanged();
-            }
+            OnPropertyChanged(nameof(GroupByInspectionType));
+            OnPropertyChanged(nameof(GroupByName));
+            OnPropertyChanged(nameof(GroupByLocation));
+            OnPropertyChanged(nameof(GroupBySeverity));
         }
 
-        private bool _groupByLocation;
-        public bool GroupByLocation
+        private bool _groupByInspection;
+        public bool GroupByInspectionType
         {
-            get => _groupByLocation;
+            get => _groupByInspection;
             set
             {
-                if (_groupByLocation == value) { return; }
-
-                if (value)
+                if (value == _groupByInspection)
                 {
-                    Results = new ObservableCollection<IInspectionResult>(
-                            Results.OrderBy(o => o.QualifiedSelection.QualifiedName.Name)
-                                .ThenBy(t => t.Inspection.Name)
-                                .ThenBy(t => t.QualifiedSelection.Selection.StartLine)
-                                .ThenBy(t => t.QualifiedSelection.Selection.StartColumn)
-                                .ToList());
+                    return;
                 }
 
-                _groupByLocation = value;
-                OnPropertyChanged();
+                _groupByInspection = value;
+                if (_groupByInspection)
+                {
+                    _groupByName = false;
+                    _groupByLocation = false;
+                    _groupBySeverity = false;
+                    SetGrouping(InspectionResultGrouping.Type);
```

Comment:
It seems to me that we can further cut down the amount of code and eliminate human errors by using a bitmask rather than 4 booleans. That would make it more consistent with filters which already uses a bitmask, as well.

Reply:
Agreed.  That's on the agenda for the next PR on this.  I just skipped it for the time being because they're mutually exclusive, and that means making a switch in the converter.

Reply:
This is in the batch of commits I just pushed.